src/module/item/spell/index.ts

```diff
@@ -596,14 +603,13 @@ class SpellPF2e extends ItemPF2e {
         }
     }
 
-    async rollDamage(this: Embedded<SpellPF2e>, event: JQuery.ClickEvent): Promise<void> {
-        const castLevel = (() => {
-            const button = event.currentTarget;
-            const card = button.closest("*[data-spell-lvl]");
-            const cardData = card ? card.dataset : {};
-            return Number(cardData.spellLvl) || 1;
-        })();
-
+    async rollDamage(
+        this: Embedded<SpellPF2e>,
+        event: JQuery.ClickEvent<unknown, unknown, HTMLElement>
+    ): Promise<void> {
+        const slotLevel =
+            Number(event.currentTarget.closest<HTMLElement>("*[data-slot-level]")?.dataset.slotLevel) || 1;
```

Comment:
After the current changes, this should use cast level, not slot level I think. That way cantrip damage button from the spell summary can scale.